Rubrik/Public/New-RubrikMount.ps1

```diff
@@ -58,22 +60,37 @@ function New-RubrikMount
 
     TestRubrikConnection
 
+    Import-Module C:\Users\eric.chang\Documents\WindowsPowerShell\Modules\Rubrik\Public\Get-RubrikSnapshotByDate.ps1
+
     Write-Verbose -Message 'Determining which version of the API to use'
     $resources = GetRubrikAPIData -endpoint ('VMwareVMMountPost')
 
-    if (!$Date) 
-    {
-      Write-Verbose -Message 'No date entered. Taking current time.'
-      $Date = Get-Date
-    }
-
     Write-Verbose -Message 'Query Rubrik for the list of protected VM details'
     if (!$HostID)
     {
       $HostID = (Get-RubrikVM -VM $VM).hostId
     }
 
-    Write-Verbose -Message 'Query Rubrik for the protected VM snapshot list'
+    if (!$Date -And !$SnapshotID)
```

Comment:
Would be easier to make Date and SnapshotID mutually exclusive and then test to see which one the user selected. I'd suggest using a Parameter Set.